### ISSUE TITLE
Hundreds of sentry-cli*.exe Entries in /tmp

### ISSUE THREAD

**Issue by TheFrozenDuck** [getsentry/sentry-android-gradle-plugin#24]
_Android Gradle Plugin:_
**4.0.0**

_Gradle:_
**6.3.0**

_sentry gradle:_
**1.7.35**

_Optimizer:_
- [ ]  ProGuard
- [ X ]  R8

_Have you created the sentry.properties file?_
- [ X ]  Yes -> in the source root folder
- [ ]  No

_SDKs:_
- [ x ]  sentry-android -> 2.1.7
- [ ]  sentry-react-native -> which version?

---
I have the following issue:

This is  a bug with Sentry in the build system. I have a similar setup to this issue ([issue 14334](https://github.com/getsentry/sentry/issues/14334)) in another sentry repository.

Building the application leads up to 526 entries of the **sentry-cli*.exe** which takes away 8GB of memory from me. Any insight on why this is happening would be aweseome.

The description goes here ...

**Steps to reproduce:**
- build project multiple times
- find hundreds of **sentry-cli*.exe** in my /tmp folder
- try to build application, and see it crash

**Actual result:**
- hundreds of **sentry-cli*.exe**

**Expected result:**
- not hundreds of **sentry-cli*.exe** in my /tmp folder

**Comment by marandaneto** [comment]
hey @TheFrozenDuck thanks for raising this.
definitely this could be improved.

a workaround, for now, would be to install `sentry-cli` locally and add `cli.executable` to the `sentry.properties` file with the given `sentry-cli` path, this way we'd not need to extract the file which is builtin on this plugin on each run.

**Comment by marandaneto** [comment]
@cortinico can we check that we either always delete sentry-cli after executing or that we use always the same file if already there?
it should not be an issue because of https://github.com/getsentry/sentry-android-gradle-plugin/blob/main/plugin-build/src/main/kotlin/io/sentry/android/gradle/SentryCliProvider.kt#L83
it may be an issue only with the v1 Plugin but may be not with the latest beta v2

**Comment by cortinico** [comment]
> it should not be an issue because of [https://github.com/getsentry/sentry-android-gradle-plugin/blob/main/plugin-build/src/main/kotlin/io/sentry/android/gradle/SentryCliProvider.kt#L83](https://github.com/getsentry/sentry-android-gradle-plugin/blob/main/plugin-build/src/main/kotlin/io/sentry/android/gradle/SentryCliProvider.kt?rgh-link-date=2021-05-31T08%3A16%3A46Z#L83)
> it may be an issue only with the v1 Plugin but may be not with the latest beta v2

True, but the `deleteOnExit` was there also before: https://github.com/getsentry/sentry-android-gradle-plugin/blob/d11b8b380faa0918b5da1922fd5f4f57295de060/src/main/groovy/io/sentry/android/gradle/SentryPlugin.groovy#L106

I'm unable to reproduce this. I'd suggest to @TheFrozenDuck that you update the plugin to 2.x, you clean your `/tmp` and report if you still have the problem.

**Comment by marandaneto** [comment]
could be a Windows only issue as well, @TheFrozenDuck can you confirm with SGP 2.0 latest beta? thanks

**Comment by cortinico** [comment]
Were you able to reproduce in any form @TheFrozenDuck?
If not, I suggest we close this and re-open should the issue happen again.

**Comment by romtsn** [comment]
@joshafeinberg what's the location of those? And are you able to reproduce what triggers creation of new ones?

**Comment by joshafeinberg** [comment]
> @joshafeinberg what's the location of those? And are you able to reproduce what triggers creation of new ones?

Its in my temp directory and I'm not sure. it seems to be each time i build. though i just went to test it again and now i'm getting this error

```
* What went wrong:
A problem occurred starting process 'command 'C:\Users\JOSHFE~1\AppData\Local\Temp\.sentry-cli13759595853993462557.exe''
> Could not start 'C:\Users\JOSHFE~1\AppData\Local\Temp\.sentry-cli13759595853993462557.exe'
```

This happened after I killed the gradle daemon and tried to rebuild
